### PR TITLE
Sidebar refactor / style update

### PR DESCRIPTION
**Pull request by wbobeirne**
## What This Does

Addresses #145, and some general v3 parity issues

* Splits the sidebar out into multiple components
* Adds back in the promos rotator from v3
* Matches the styles and outputs as close to v3 as possible

## Still Needs Doing

* Loading states (Waiting on the PR that adds the common loader component.)
* Currency indicators (Just realized I was missing those)

## Screenshots

sidebars

### CODE REVIEW

**Comment by wbobeirne** [comment]
I may circle back around and tighten up the spacing, seeing the two next to each other v4 looks a lot spacier than v3.

**Comment by wbobeirne** [comment]
Oh boy, this commit history looks like garbage. I had merged my style PR in here to get working before it landed on develop. We should probably squash this merge commit.

**Comment by dternyak** [comment]
@wbobeirne Will squash 👍

File: common/components/BalanceSidebar/TokenBalances/AddCustomTokenForm.jsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    }
+    if (!symbol) {
+      errors.symbol = true;
+    }
+
+    return errors;
+  }
+
+  isValid() {
+    console.log(Object.keys(this.getErrors()).length);
```
Comment (dternyak) [inline common/components/BalanceSidebar/TokenBalances/AddCustomTokenForm.jsx]: 🌲

File: common/components/BalanceSidebar/AccountInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import React from 'react';
+import translate from 'translations';
+import { Identicon } from 'components/ui';
+import { formatNumber } from 'utils/formatters';
+import type Big from 'bignumber.js';
+import type { BaseWallet } from 'libs/wallet';
+import type { NetworkConfig } from 'config/data';
+
+type Props = {
+  balance: Big,
```
Comment (dternyak) [inline common/components/BalanceSidebar/AccountInfo.jsx]: I have balances / anything related to units typed out in `use-unit-types`. I'll probably go back through and convert this (and anything else this PR introduces) before PRing `use-unit-types`.

File: common/components/BalanceSidebar/EquivalentValues.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import React from 'react';
+import translate from 'translations';
+import { Link } from 'react-router';
+import { formatNumber } from 'utils/formatters';
+import type Big from 'bignumber.js';
+
+const ratesKeys = ['BTC', 'REP', 'EUR', 'USD', 'GBP', 'CHF'];
+
+type Props = {
+  balance: Big,
```
Comment (dternyak) [inline common/components/BalanceSidebar/EquivalentValues.jsx]: Not for me to use Ether type here later.

File: common/components/BalanceSidebar/index.jsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import { showNotification } from 'actions/notifications';
+
+import AccountInfo from './AccountInfo';
+import Promos from './Promos';
+import TokenBalances from './TokenBalances';
+import EquivalentValues from './EquivalentValues';
+
+type Props = {
+  wallet: BaseWallet,
+  balance: Big,
```
Comment (dternyak) [inline common/components/BalanceSidebar/index.jsx]: Also here

File: common/components/BalanceSidebar/Promos.jsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    ]
+  },
+  {
+    color: '#2b71b1',
+    href:
+      'https://buy.coinbase.com?code=a6e1bd98-6464-5552-84dd-b27f0388ac7d&address=0xA7DeFf12461661212734dB35AdE9aE7d987D648c&crypto_currency=ETH&currency=USD',
+    isExternal: true,
+    texts: [
+      <p key="1">It’s now easier to get more ETH</p>,
+      <h5 key="2">Buy ETH with USD</h5>
```
Comment (HenryNguyen5) [inline common/components/BalanceSidebar/Promos.jsx]: When i was working on ENS general info list, I found it pretty weird having to manually assign keys in an array literal, maybe we should eventually figure out a solution around this in the future

**Pull request merged**